src/middlewared/middlewared/plugins/vm.py

```diff
@@ -1594,6 +1606,10 @@ def status(self, id):
 
     @private
     def close_libvirt_connection(self):
+        if self.libvirt_event_loop_thread:
```

Comment:
Related to my comment  below about `setup_libvirt_events`: if you refactor this out to `teardown_libvirt_events`, you'll be able to hide these three variables into events service part.